File: PrezBDDTDD.Tests.Node/src/service/ServiceVirement.js
var VirementStatus = require('../model/VirementStatus');

module.exports = class ServiceVirement {
    constructor(plafond){
        this.plafond = plafond;
    }

    effectuerVirement(montant, compteDebiteur, compteCrediteur) {
        if (montant > compteDebiteur.solde)
        {
            return VirementStatus.SOLDE_DEPASSE;
        }

        if (montant > this.plafond)
        {
            return VirementStatus.PLAFOND_DEPASSE;
        }

        compteDebiteur.solde -= montant;
        compteCrediteur.solde += montant;
        return VirementStatus.OK;
    }
};

File: PresBDDTDD.Tests.IOSSwift/PrezBDDTDD/PrezBDDTDD/Service/ServiceVirement.swift
//
//  ServiceVersement.swift
//  PrezBDDTDD
//
//  Created by Yann Breleur on 28/02/2017.
//  Copyright © 2017 AXA. All rights reserved.
//


class ServiceVirement {
    let plafond:Int?
    
    init(_ plafond:Int?) {
        self.plafond = plafond
    }
    
    func effectuerVirement(de montant: Int, duCompte compteDebiteur: Compte, auCompte compteCrediteur: Compte) -> VirementStatus {
        
        if montant > compteDebiteur.solde {
            return VirementStatus.SOLDE_DEPASSE
        }
        
        if let plafond = plafond,  montant > plafond{
            return VirementStatus.PLAFOND_DEPASSE
        }
        compteDebiteur.solde -= montant
        compteCrediteur.solde += montant
        return VirementStatus.OK
    }
}
